### PR TITLE
Add FlushAsync cancellation support

### PR DESCRIPTION
**Pull request by pakrym**
@halter73 @davidfowl

### CODE REVIEW

File: src/System.IO.Pipelines/Pipe.cs [hunk 1]
```diff
@@ -377,6 +377,11 @@ void IPipeWriter.Complete(Exception exception)
             }
         }
 
+        public void CancelPendingFlush()
+        {
+            throw new NotImplementedException();
```
Comment (davidfowl) [inline src/System.IO.Pipelines/Pipe.cs]: Wut?

Comment (pakrym) [inline src/System.IO.Pipelines/Pipe.cs]: How did it pass tests...

Comment (pakrym) [inline src/System.IO.Pipelines/Pipe.cs]: Oh

File: tests/System.IO.Pipelines.Tests/BackpressureTests.cs [hunk 2]
```diff
@@ -110,7 +112,39 @@ public void FlushAsyncReturnsFalseIfReaderCompletes()
...
+            var writableBuffer = _pipe.Writer.Alloc(64);
+            writableBuffer.Advance(64);
+            var flushAsync = writableBuffer.FlushAsync();
+
+            Assert.False(flushAsync.IsCompleted);
+
+            _pipe.Writer.CancelPendingFlush();
+
+            Assert.True(flushAsync.IsCompleted);
+            var flushResult = flushAsync.GetResult();
```
Comment (davidfowl) [inline tests/System.IO.Pipelines.Tests/BackpressureTests.cs]: This isn't used.

File: tests/System.IO.Pipelines.Tests/BackpressureTests.cs [hunk 3]
```diff
@@ -110,7 +112,39 @@ public void FlushAsyncReturnsFalseIfReaderCompletes()
...
+        {
+            var writableBuffer = _pipe.Writer.Alloc(64);
+            writableBuffer.Advance(64);
+            var flushAsync = writableBuffer.FlushAsync();
+
+            Assert.False(flushAsync.IsCompleted);
+
+            _pipe.Writer.CancelPendingFlush();
+
+            Assert.True(flushAsync.IsCompleted);
```
Comment (davidfowl) [inline tests/System.IO.Pipelines.Tests/BackpressureTests.cs]: What does IsCompleted mean?

**Comment by davidfowl** [comment]
@dotnet-bot test this please

**Pull request merged**